### PR TITLE
Handle annotation import with enable AUTO_IMPORT_NAMES not changed (for non-namespaced file)

### LINKED ISSUE #7051: PHPDoc Annotation not changed to imported class
### Discussed in https://github.com/rectorphp/rector/discussions/7049

<div type='discussions-op-text'>

<sup>Originally posted by **dahaupt** March 14, 2022</sup>
I've noticed a different behavior after updating to v0.12.17.

Rector adds a new use statement:
```diff
+use TYPO3\CMS\Extbase\Annotation\Validate;
```

But does not change the related PHPDoc annotation:
```php
/**
* @TYPO3\CMS\Extbase\Annotation\Validate
*/
public function method(){}
```

PHP-CS-Fixer afterwards removes the unused import. 

I am not sure if this is specific to TYPO3 or a configuration mistake of myself. Thankful for any help!</div>

---

Reproduction repository: https://github.com/dahaupt/rector-annotation-import

### PR DESCRIPTION
**Pull request by samsonasik**
Given the following code:

```php
/**
 * @Doctrine\ORM\Mapping\Table("Table_Name")
 */
class NoNamespace
{
}
```

It transformed to:

```diff
+use Doctrine\ORM\Mapping\Table;
 /**
  * @Doctrine\ORM\Mapping\Table("Table_Name")
  */
```

which the annotation should be renamed to:

```diff
+use Doctrine\ORM\Mapping\Table;
 /**
-  * @Doctrine\ORM\Mapping\Table("Table_Name")
+  * @Table("Table_Name")
  */
```

as Table imported.

This PR try to Fixes https://github.com/rectorphp/rector/issues/7051

### CODE REVIEW

File: tests/Issues/AnnotationImport/Fixture/skip_with_namespace_not_exists.php.inc [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+namespace App;
+
+/**
+ * @Doctrine\ORM\Mapping\Table("Table_Name")
```
Comment (samsonasik) [inline tests/Issues/AnnotationImport/Fixture/skip_with_namespace_not_exists.php.inc]: for namespaced class, eg, has `namespace App;`, if it going to be imported, it should be with `App` prefix: `App\Doctrine\ORM\Mapping\Table`, so I think it should be skipped instead.

**Comment by samsonasik** [comment]
I am not sure about inner namespaced annotation and annotation with `\` prefix, currently this PR only handle on non-namespaced file with namespaced annotation.

File: tests/Issues/AnnotationImport/Fixture/skip_with_namespace_not_exists.php.inc (hunk 1 above)
Comment (samsonasik) [reply tests/Issues/AnnotationImport/Fixture/skip_with_namespace_not_exists.php.inc]: fixture removed for now so this PR only handle based on reported issue https://github.com/rectorphp/rector/issues/7051 for non-namespaced file.

**Comment by samsonasik** [comment]
All checks have passed 🎉 @TomasVotruba it is ready for review.

**Comment by TomasVotruba** [comment]
Thank you, well done :+1:

**Pull request merged**